### PR TITLE
Bug 1797512: Reporting RequestFailure errors

### PR DESCRIPTION
**Pull request by ricardomaraschini**
If we fail to reach AWS endpoint (or custom region endpoint) report this
clearly on operator conditions.

### CODE REVIEW

**Comment by ricardomaraschini** [comment]
/test e2e-aws

"Failed to import expected imagestreams"

File: pkg/storage/s3/s3.go [hunk 1]
```diff
@@ -369,7 +369,11 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 				Bucket: aws.String(d.Config.Bucket),
 			})
 			if err != nil {
-				if aerr, ok := err.(awserr.Error); ok {
+				if aerr, ok := err.(awserr.RequestFailure); ok {
+					err := fmt.Errorf("error requesting endpoint: %s(%d)", aerr.Code(), aerr.StatusCode())
```
Comment (dmage) [inline pkg/storage/s3/s3.go:373]: I would rather not fix the bug than lose the error message.

Comment (ricardomaraschini) [reply pkg/storage/s3/s3.go:373]: Let's log it?

**Comment by ricardomaraschini** [comment]
/close

We have decided to keep the complete error being reported on Conditions.

**Pull request closed without merging**